### PR TITLE
Issue61 73 98 - Project Build Update to Include Pyproject.Toml and Docker

### PR DESCRIPTION
**Pull request by PatrickWeller**
### Summary
- [x] Pull request has a descriptive name
- [x] Briefly describe the purpose of this pull request.
- [x] Link any relevant issue numbers (e.g., `Fixes #123`).
- [x] Add the milestone of the sprint

### Code Changes
- [x] The code changes are clearly described
- [not relevant] Code is PEP8 compliant 
    (Pylint can be used to show real time issues, Black can be used to enforce)
- [x] Are only the relevant files included in the commits (e.g. no pycache or log file)

### Documentation
- [x] Are there sufficient in line comments?
- [ not relevant ] Is there a docstring for each function or class in Numpy format
- [ not relevant ] Is there top level documentation in the file and is it up to date
- [x] Has the README been updated if relevant

### Dependencies
- [ ] Are there any new dependencies?
  - I mean docker is needed, but that's outside the env?
- [x] Have the dependencies been added to the requirements file or environment file with version?

### Testing
- [x] This PR has been tested locally.
- [ ] Automated tests have been added or updated as needed.
  - maybe I need to add tests for this but please not now
- [x] Tests pass without errors.
- [not relevant] Logging statements added (if it's a core script)

### Review and Approval
- [x] Self-review completed.
- [x] Ready for team review.
## Summary ##

**Docker Integration:**
Added a Dockerfile to enable containerization of the project. #98 #61
Created an entrypoint.sh script for activating the Conda environment within the container.

**Licensing:**
Added an MIT license with details on third-party dependencies. #75

**Environment & Build Configuration:**
Moved environment.yaml to the root directory and removed requirements.txt.
Introduced pyproject.toml for modern Python packaging and dependency management. #73 #61

**Documentation:**
Updated README.md to reflect the addition of Docker support and clarified usage instructions.
Added licensing details to the README. #75

**Workflow Updates:**

Fixed references in .github/workflows/automated_tests.yml to ensure the correct paths for the Conda environment and cache key after environment file moved.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -17,29 +17,29 @@ PanelPal is a python package of command line tools for helping UK labs implement
...
    git clone https://github.com/PatrickWeller/PanelPal.git
     ```
 
-2. Create a new conda environment with the following command:
+2. Build the docker image:
 
    ```bash
-   conda env create -f env/environment.yaml
+   cd PanelPal
+   docker build -it panelpal .
```
Comment (ASendellPrice) [inline README.md:30]: typo, should be:
docker build -t panelpal .

**Comment by ASendellPrice** [comment]
Updated code has been reviewed and tested on my laptop. I was able to build the docker container and run PanelPal generate-bed from within it. Only changes that need implementing prior to merge is updating the readme to resolve two typos identified. Great work @PatrickWeller

File: README.md (hunk 1 above)
Comment (PatrickWeller) [reply README.md]: Corrected in latest commit

File: README.md [hunk 2]
```diff
@@ -17,29 +17,29 @@ PanelPal is a python package of command line tools for helping UK labs implement
...
     ```
 
-3. Activate the environment:
+3. Run the docker container:
 
     ```bash
-    conda activate PanelPal
+    docker run -t panelpal
```
[the remark this answers is not in the document]
Comment (PatrickWeller) [reply README.md]: Corrected in latest commit

**Comment by PatrickWeller** [comment]
@ASendellPrice Changes made to requirements.txt and pyproject.toml, conflicts resolved. Please check and provided the tests still run OK you can merge. Thanks.

**Comment by ASendellPrice** [comment]
All tests still working. During testing I identified a bug with generate_bed.py (see issue #111). Bug has been fixed during this PR.

Ready to merge into development.

**Pull request merged**